ppl/cmd/zident/user/user.go

```diff
@@ -0,0 +1,232 @@
+package user
+
+import (
+	"context"
+	"crypto/rand"
+	"encoding/base64"
+	"errors"
+	"flag"
+	"fmt"
+	"net/url"
+	"os"
+	"strings"
+
+	"github.com/brimsec/zq/ppl/cmd/zident/root"
+	"github.com/go-resty/resty/v2"
+	"github.com/mccanne/charm"
+	"github.com/segmentio/ksuid"
+	"go.uber.org/multierr"
+	"gopkg.in/auth0.v5/management"
+)
+
+const auth0DefaultConnection = "Username-Password-Authentication"
+
+type auth0ClientConfig struct {
+	clientId     string
+	clientSecret string
+	connection   string
+	domain       url.URL
+}
+
+func (c *auth0ClientConfig) FromEnv() error {
+	var err error
+	if c.clientId = os.Getenv("AUTH0_CLIENT_ID"); c.clientId == "" {
+		err = multierr.Append(err, errors.New("AUTH0_CLIENT_ID not set"))
+	}
+	if c.clientSecret = os.Getenv("AUTH0_CLIENT_SECRET"); c.clientSecret == "" {
+		err = multierr.Append(err, errors.New("AUTH0_CLIENT_SECRET not set"))
+	}
+	if c.connection = os.Getenv("AUTH0_CONNECTION"); c.connection == "" {
+		err = multierr.Append(err, errors.New("AUTH0_CONNECTION not set"))
+	}
+	domain := os.Getenv("AUTH0_DOMAIN")
+	if domain == "" {
+		err = multierr.Append(err, errors.New("AUTH0_DOMAIN not set"))
+	} else {
+		u, err := url.Parse(domain)
+		if err != nil {
+			err = multierr.Append(err, errors.New("AUTH0_DOMAIN is not a URL"))
+		} else {
+			c.domain = *u
+		}
+	}
+	if err != nil {
+		return fmt.Errorf("unable to configure auth0 client: %w", err)
+	}
+	return nil
+}
+
+type searchFlags struct {
+	email    string
+	tenantID string
+	userID   string
+}
+
+func (f *searchFlags) SetFlags(fs *flag.FlagSet) {
+	fs.StringVar(&f.email, "email", "", "search by email address")
+	fs.StringVar(&f.tenantID, "tenantid", "", "search by tenant ID")
+	fs.StringVar(&f.userID, "userid", "", "search by user ID")
+}
+
+func findUser(ctx context.Context, m *management.Management, f searchFlags) (*management.User, error) {
+	var user *management.User
+	if err := streamUsers(ctx, m, f, false, func(u *management.User) error {
+		user = u
+		return nil
+	}); err != nil {
+		return nil, err
+	}
+	if user == nil {
+		return nil, errors.New("no user found")
+	}
+	return user, nil
+}
+
+func streamUsers(ctx context.Context, m *management.Management, f searchFlags, allowMultiple bool, fn func(*management.User) error) error {
```

Comment:
"management" is kind of a vague name for the main Auth0 interface. Now that I see it, I will expect it in this context, so NP.